### ISSUE TITLE
ImportError: cannot import name 'HeatSourceTypes' from 'carrier_api.const'

### ISSUE THREAD

**Issue by FiveElementNinja** [dahlb/ha_carrier#174]
## System Information

version | core-2024.12.5
-- | --
installation_type | Unknown
dev | false
hassio | false
docker | false
user | root
virtualenv | false
python_version | 3.13.0
os_name | Linux
os_version | 6.1.21-v8+
arch | aarch64
timezone | America/Chicago
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
HACS Data | ok
GitHub API Calls Remaining | 4904
Installed Version | 2.0.1
Stage | running
Available Repositories | 1492
Downloaded Repositories | 11

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Dashboards</summary>

dashboards | 8
-- | --
resources | 3
views | 8
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | December 20, 2024 at 9:54 PM
-- | --
current_recorder_run | December 31, 2024 at 12:09 PM
estimated_db_size | 438.90 MiB
database_engine | sqlite
database_version | 3.45.3

</details>

### Checklist

- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of currently [previous issues](https://github.com/dahlb/ha_carrier/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

Ever since `v1.27.0` I have been receiving this error and the entity fails to load. This occurs in every version starting from that version. `v1.26.2` is the last working version for me, so that's what I'm stuck on for the time being.

`ImportError: cannot import name 'HeatSourceTypes' from 'carrier_api.const' (/usr/local/lib/python3.13/site-packages/carrier_api/const.py)`

Additionally, I also receive this warning, which I assume is due to the error.

```Logger: homeassistant.util.loop
Source: util/loop.py:136
First occurred: 12:09:32 PM (1 occurrences)
Last logged: 12:09:32 PM

Detected blocking call to import_module with args ('custom_components.ha_carrier.select',) inside the event loop by custom integration 'ha_carrier' at custom_components/ha_carrier/__init__.py, line 80: await hass.config_entries.async_forward_entry_setups(config_entry, PLATFORMS) (offender: /usr/src/homeassistant/homeassistant/loader.py, line 1303: return importlib.import_module(f"{self.pkg_path}.{platform_name}")), please create a bug report at https://github.com/dahlb/ha_carrier/issues For developers, please see https://developers.home-assistant.io/docs/asyncio_blocking_operations/#import_module Traceback (most recent call last): File "<frozen runpy>", line 198, in _run_module_as_main File "<frozen runpy>", line 88, in _run_code File "/usr/src/homeassistant/homeassistant/__main__.py", line 227, in <module> sys.exit(main()) File "/usr/src/homeassistant/homeassistant/__main__.py", line 213, in main exit_code = runner.run(runtime_conf) File "/usr/src/homeassistant/homeassistant/runner.py", line 154, in run return loop.run_until_complete(setup_and_run_hass(runtime_config)) File "/usr/local/lib/python3.13/asyncio/base_events.py", line 708, in run_until_complete self.run_forever() File "/usr/local/lib/python3.13/asyncio/base_events.py", line 679, in run_forever self._run_once() File "/usr/local/lib/python3.13/asyncio/base_events.py", line 2027, in _run_once handle._run() File "/usr/local/lib/python3.13/asyncio/events.py", line 89, in _run self._context.run(self._callback, *self._args) File "/usr/src/homeassistant/homeassistant/config_entries.py", line 788, in async_setup_locked await self.async_setup(hass, integration=integration) File "/usr/src/homeassistant/homeassistant/config_entries.py", line 551, in async_setup await self.__async_setup_with_context(hass, integration) File "/usr/src/homeassistant/homeassistant/config_entries.py", line 640, in __async_setup_with_context result = await component.async_setup_entry(hass, self) File "/config/custom_components/ha_carrier/__init__.py", line 80, in async_setup_entry await hass.config_entries.async_forward_entry_setups(config_entry, PLATFORMS)
```

### Reproduction steps

1. Install v1.27.0 or higher
2. Check the logs for the error
3. Observe that the entity hasn't loaded and therefore the card in the dashboard doesn't load.
...

### Debug logs

```text
Logger: homeassistant.config_entries
Source: config_entries.py:640
First occurred: 12:09:32 PM (1 occurrences)
Last logged: 12:09:32 PM

Error setting up entry [REDACTED] for ha_carrier
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 640, in __async_setup_with_context
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/ha_carrier/__init__.py", line 80, in async_setup_entry
    await hass.config_entries.async_forward_entry_setups(config_entry, PLATFORMS)
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 2278, in async_forward_entry_setups
    await integration.async_get_platforms(platforms)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1184, in async_get_platforms
    import_future.result()
    ~~~~~~~~~~~~~~~~~~~~^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1184, in async_get_platforms
    import_future.result()
    ~~~~~~~~~~~~~~~~~~~~^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1184, in async_get_platforms
    import_future.result()
    ~~~~~~~~~~~~~~~~~~~~^^
  [Previous line repeated 1 more time]
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1172, in async_get_platforms
    platforms.update(self._load_platforms(platform_names))
                     ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1097, in _load_platforms
    platform_name: self._load_platform(platform_name)
                   ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1271, in _load_platform
    cache[full_name] = self._import_platform(platform_name)
                       ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1303, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
           ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 200, in protected_loop_func
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.13/importlib/__init__.py", line 88, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 1022, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/ha_carrier/select.py", line 7, in <module>
    from carrier_api.const import HeatSourceTypes
ImportError: cannot import name 'HeatSourceTypes' from 'carrier_api.const' (/usr/local/lib/python3.13/site-packages/carrier_api/const.py)
```

**Comment by dahlb** [comment]
this error message implies your home assistant environment is not managing dependencies correctly

if you are hosting ha on a linux distro be sure you install carrier_api v[1.16.2](https://github.com/dahlb/carrier_api/commit/a778d1b7ccebe11680a39f8bc6f0bcce0ec3ca56) if you are using home assistant os or the official docker image this would be managed for you.

**Comment by FiveElementNinja** [comment]
Thanks for the tip.

I am running my instance in Docker on Debian. You're right that the Docker image should be managing it for me, but ever since the December update to HA it stopped installing/updating dependencies automatically, so I have to do them manually. Previously I was able to just do `pip install carrier_api` and that would fix it but it seems that this time I had to tell it to force reinstall. Once I did that, everything started working again.

Sorry to bother. Thanks again.
